src/main/java/com/lithium/dbi/rdbi/recipes/cache/RedisMultiCache.java

```diff
@@ -0,0 +1,146 @@
+package com.lithium.dbi.rdbi.recipes.cache;
+
+import com.google.common.base.Function;
+import com.google.common.base.Predicates;
+import com.google.common.collect.FluentIterable;
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableMap;
+import com.google.common.collect.Sets;
+import com.lithium.dbi.rdbi.Handle;
+import com.lithium.dbi.rdbi.RDBI;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.annotation.Nullable;
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+import static com.google.common.base.MoreObjects.firstNonNull;
+
+/**
+ * A loading cache for apis that prefer to load multiple keys in a single round trip.
+ * <p/>
+ * Contrast this cache with {@link RedisHashCache}, which is also a hash-based redis-backed cache.  Whereas {@link RedisHashCache}
+ * fetches data for missed cached keys iteratively, this {@link RedisMultiCache} expects to resolve multiple cache misses with a single
+ * call to a bulk loader.  Moreover, this cache makes no assumptions about being able to load values for all of the caches potential keys.
+ * <p/>
+ * This cache is <b>not</b> an automatically refreshing cache.  After a key expires, the next request for the key may
+ * incur some loading penalty.
+ */
```

Comment:
I'm reluctant to generalize as I feel strongly that there are two separate uses cases supported.  Let me describe further to see whether you agree?  

RedisHashCache (existing)
 - 10s - 100s of valid keys
 - we'll often want _all_ the values and sometimes want just one
 - auto-refreshing, which makes for a relatively small key space
 - example: all of the analytics dashboard definitions

RedisMultiCache (new)
 - 100s to 1,000s of valid keys
 - we never want _all_ of the values; rather, we'll ask for a small subset of the possible keys
 - auto-expiring; shouldn't auto-refresh because universe is too large 
 - example: tocoma asset details for an analytics report

What do you think?